misk/src/main/kotlin/misk/web/interceptors/RequestLoggingInterceptor.kt

```diff
@@ -24,7 +24,7 @@ private val logger = getLogger<RequestLoggingInterceptor>()
  * Logs request and response information for an action.
  * Timing information doesn't count time writing the response to the remote client.
  */
-class RequestLoggingInterceptor internal constructor(
+class RequestLoggingInterceptor(
```

Comment:
There’s a ton of parameters here, and I pity the poor code that has to call this constructor. I also dislike the premise of future maintainers introducing new parameters here and breaking you.

Would it work equivalently well to create an overload of `Factory.create()` that takes the parameters you need, but otherwise gets Guice to do the work for the rest? Then your code injects a Factory and calls create() rather than calling this constructor directly?

Reply:
Thinking about the use case presented here. I think it speak on how the current way to configure (compile time) is not enough. Maybe we should a mechanism to override that using the config files?

Reply:
The `create` method has to conform to `NetworkInterceptor.Factory.create`, since that's what is going to get called [in WebActionFactory](https://github.com/cashapp/misk/blob/c2cd9a13b0f4dbe0b6e7c35c9883487f65aff183/misk/src/main/kotlin/misk/web/actions/WebActionFactory.kt#L204).

That said, I could just change this PR to instead add [the factory that I'm adding to Plasma](https://github.com/squareup/cash-plasma/blob/a855a5e562541e9e4cbdb30682f53e1304b6b7f8/client/src/main/kotlin/com/squareup/plasma/logging/LogPlasmaRequestResponse.kt#L31). It seems generally reusable; it's just a factory that uses a single (injected) global config instead of relying on config from its annotation.

Reply:
https://github.com/cashapp/misk/pull/2726